language/move-prover/spec-lang/src/env.rs

```diff
@@ -2256,16 +2256,29 @@ impl<'env> FunctionEnv<'env> {
             .get_function_source_map(self.data.def_idx)
         {
             if let Some((ident, _)) = fmap.get_parameter_or_local_name(idx as u64) {
-                // The Move compiler produces temporay names of the form `<foo>%#<num>`.
-                // Ignore those names and use the idx-based repr instead.
-                if !ident.contains("%#") {
-                    return self.module_env.env.symbol_pool.make(ident.as_str());
-                }
+                // The Move compiler produces temporary names of the form `<foo>%#<num>`,
+                // where <num> seems to be generated non-deterministically.
```

Comment:
I was just fairly lazy in generating the new temps as it didn't matter at the time. 

I do map them back to the original name for the spec exp blocks. Do we need to do the same here? or does it not matter too much?

Reply:
We only need to remap those because we need deterministic output for baseline files. Baseline files can contain references to those names for negative prover tests.

Reply:
The more I think about it, I'm surprised this isn't deterministic already. Do you have repro I can look at?

Reply:
The best I can imagine is in a clean client, uncomment this special treatment here and then run the prover tests multiple times. You'll see over time baseline failures because the number here as changed.

In any case, this is not related to this PR (this treatment was there since long, I only changed a bit of the logic unrelated to what we are discussing).